gatsby/content/docs/e2ee_cross_signing.mdx

```diff
@@ -1196,3 +1196,223 @@ and use that to verify and sign yourself:
+Creating the key information for the new `m.secret_storage.key.<keyid>` event is basically just doing
+what you did before to verify if an entered key is correct, but now using the data to set the account
+data event. In the case of a passphrase you will have to generate a salt and iteration times yourself.
+All keys have 256 bits.
+
+**Important**: Make sure that all random things generated here, be it keys or salts, are as securely
+random as possible. Many languages have libraries for creating more securely random data than the
+default pseudo random number generator.
+
+#### Key from passphrase
+A reasonably-secure iteration number for PBKDF2 is `500000` (That is five-hundred thousand). The salt
+needs as much random data as our hash function. While PBKDF2 uses sha512 as hash, the generated key
+is only 256 bits long, so 256 bits should be reasonably secure. To make sure that the salt is usable,
+you can just base64 the randomly generated salt.
+
+After that, you use PBKDF2 to get the key from the passphrase, and calculate the MAC and IV.
+
+Code for generating the key could look as follows:
+```dart
+// "passphrase" is the passphrase by the user
+
+var passphraseInfo = {
+  "algorithm": "m.pbkdf2",
+  "iterations": 500000,
+  "salt": base64.encode(SecureRandom(32).bytes), // we use secure random! 32 bytes = 256 bits
+  "bits": 256,
+};
+var generator = PBKDF2(hashAlgorithm: sha512);
+var key = Uint8List.fromList(generator.generateKey(passphrase, passphraseInfo["salt"],
+      passphraseInfo["iterations"], passphraseInfo["bits"] / 8));
+```
+
+#### Generating a random key without a passphrase
+Just generate a 256-bit long key.
+
+```dart
+var key = SecureRandom(32).bytes; // 32 bytes = 256 bits
+```
+
+#### Calculating the key information
+You still need an IV to be able to calculate the MAC you later on use to verify the key. For that,
+generate 16 bytes and base64-encode them. Afterwards you just generate the MAC like you would for
+verifying the key, and then store the account data event.
+
+```dart
+// "key" is the key
+var iv = base64.encode(SecureRandom(16).bytes);
+// ZERO_STR are 32 bytes of zero. We encrypt with our generated key, a blank name and the iv of the event
+var encrypted = encryptAes(ZERO_STR, key, '', iv);
+var info = {
+  "algorithm": "m.secret_storage.v1.aes-hmac-sha2",
+  "iv": iv,
+  "mac": encrypted.mac,
+  "passphrase": passphraseInfo,
+};
+```
+
+#### Generating the key ID
+The key ID is just a short, random string which doesn't contain a dot. For that, you can e.g. generate
+a few random bytes and base64 encode them. Make sure no key with such an ID exists yet! After you have
+the key ID, set your key information object to account data `m.secret_storage.key.<keyId>`.
+
+To set the newly generated key as default key, simply set the `m.secret_storage.default_key` account
+data to the key ID (*NOT* the full account data type):
+
+```json
+{
+  "key": "<keyId>"
+}
+```
+
+### Bootstrappning cross-signing
```

Comment:
```suggestion
### Bootstrapping cross-signing
```

Reply:
@Sorunome Mind accepting this change? (I don't have access for your fork)

Reply:
Fixed typo in https://github.com/matrix-org/matrix.org/commit/cd572b2eaa9d4b2d19fa78aa99c37e5c5c0a848c